### ISSUE TITLE
amplify mock api not working with subscriptions

### ISSUE THREAD

**Issue by pieter-vandenburg** [aws-amplify/amplify-category-api#2221]
### How did you install the Amplify CLI?

npm install -g @aws-amplify/cli

### If applicable, what version of Node.js are you using?

v21.5.0

### Amplify CLI Version

12.10.1

### What operating system are you using?

Arch Linux

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No

### Describe the bug

When using mock for the backend

`$ amplify mock api`

With the following subscription:

```javascript
useEffect(() => {
  const subscription = client
    .graphql({ query: subscriptions.onCreateTodo })
    .subscribe({
      next: ({ data }) => console.log(data),
      error: (error) => console.warn(error)
    });
  return () => subscription.unsubscribe();
}, []);
```

And the following schema:

```graphql
type Todo @model {
  id: ID!
  name: String!
  description: String
}
```

fails with error:

```text
index.ts:69 WebSocket connection to 'wss://192.168.1.10:20002/graphql/realtime?header=LONG_CODE_TRUCATED&payload=e30=' failed: 
```

### Expected behavior

Testing api locally with mock should work. I've tried this in other projects with differing schema's but all fail with the same error. Broke this problem down to it's basics for bug report.

### Reproduction steps

1. `npx create-react-app test_amplify`
2. `cd test_amplify`
3. `amplify init`
4. `amplify add api` (Accepting defaults)
5. `npm install aws-amplify`
6. Write the subscription code above in `App.js`
7. `amplify mock api`
8. `npm run start`

### Project Identifier

2ebfbe4ebc7ed43fb28d4bdadd09d8be

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

Similiar to issue [555](https://github.com/aws-amplify/amplify-category-api/issues/555) and [122](https://github.com/aws-amplify/amplify-category-api/issues/122) but different output and they are resolved with version updates. Also, I'm not sure if the local protocol should be `ws://` rather than `wss://`, but I thought it worth pointing out.

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by pieter-vandenburg** [comment]
![image](https://github.com/aws-amplify/amplify-category-api/assets/32934955/42e9bcef-c588-48b0-91e5-fe9c12e8cfa7)

**Comment by pieter-vandenburg** [comment]
Is there any movement on this?

**Comment by AakashKB** [comment]
I've been emailing them on same issue, seems it is blocked by another PR
